pkg/scheduler/frameworkext/controllers.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package frameworkext
+
+import (
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/descheduler/framework"
+)
+
+type ControllerProvider interface {
+	NewControllers() ([]Controller, error)
+}
+
+type Controller interface {
+	Start()
+	Name() string
+}
+
+type ControllersMap struct {
+	controllers map[string]Controller
+}
+
+func NewControllersMap() *ControllersMap {
+	return &ControllersMap{
+		controllers: make(map[string]Controller),
+	}
+}
+
+func (cm ControllersMap) RegisterControllers(plugin framework.Plugin) {
+	if controllerProvider, ok := plugin.(ControllerProvider); ok {
+		if controllers, err := controllerProvider.NewControllers(); err == nil {
```

Comment:
We should ensure that the `NewControllers` method of only one plugin is called at a time to avoid some unexpected problems.